src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java

```diff
@@ -175,6 +175,8 @@ public BlocklyPanel(MCreator mcreator) {
 					// register JS bridge
 					JSObject window = (JSObject) webEngine.executeScript("window");
 					window.setMember("javabridge", bridge);
+					if (type != null)
+						window.setMember("editorType", type.registryName());

```

Comment:
I guess this new field could/should be mandatory, right?

Reply:
Could be, you can mark it as NonNull and remove the check